zman-backend/src/main/kotlin/com/peterservice/zman/controllers/AuthenticationInformationController.kt

```diff
@@ -0,0 +1,24 @@
+package com.peterservice.zman.controllers
+
+import com.peterservice.zman.api.entities.Authentication
+import org.springframework.http.HttpStatus
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import javax.servlet.http.HttpServletRequest
+
+@RestController
+class AuthenticationInformationController {
+    @GetMapping("/authentication")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    fun authentication(request: HttpServletRequest): Authentication {
+        if (request.userPrincipal == null) {
+            return Authentication("anonymousUser", false)
```

Comment:
for non-authenticated communication there is no need to have a "user" in Authentication object